staging/src/k8s.io/metrics/pkg/client/custom_metrics/client.go

```diff
@@ -85,17 +79,11 @@ func (c *customMetricsClient) NamespacedMetrics(namespace string) MetricsInterfa
 }
 
 func (c *customMetricsClient) qualResourceForKind(groupKind schema.GroupKind) (string, error) {
-	mapping, err := c.mapper.RESTMapping(groupKind)
-	if err != nil {
-		return "", fmt.Errorf("unable to map kind %s to resource: %v", groupKind.String(), err)
-	}
-
-	groupResource := schema.GroupResource{
-		Group:    mapping.GroupVersionKind.Group,
-		Resource: mapping.Resource,
-	}
-
-	return groupResource.String(), nil
+	// the version doesn't matter
+	gvk := groupKind.WithVersion("")
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
```

Comment:
This is suboptimal, IMO.  We should continue allowing a mapper to be passed in, so that we can use discovery information to get this transformation.